models/repo.go

```diff
@@ -1326,7 +1326,19 @@ func createRepository(e *xorm.Session, doer, u *User, repo *Repository) (err err
 			return fmt.Errorf("getOwnerTeam: %v", err)
 		} else if err = t.addRepository(e, repo); err != nil {
 			return fmt.Errorf("addRepository: %v", err)
-		} else if err = prepareWebhooks(e, repo, HookEventRepository, &api.RepositoryPayload{
+		}
+
+		if !t.IsMember(doer.ID) {
```

Comment:
shouldn't we also check if user is part of any team with admin rights?

Reply:
It will only be possible if #6791 gets merged. I am touching the same code so the order will be important here. (The new code in #6791 for adding global teams has to be placed before checking admin access). So I don't know if I should try to solve it now?

Reply:
Imho it would be better if this checkbox would be allowed only for teams with right level admin and a repo would be added to that team instead of setting individual rights for creator

Reply:
That would restrict the usage more. 
In my company's use case we want most users to be able to create repositories in the organization. But they shall definitely not be owners. And there are no pre-defined set of users or teams that shall have access to a new repo (and specifically not as repo admins,). However if a user create a repo it is ok that he have admin access to it (at least initially).

If #6791 is implemented you can have a team with global admin access and create repo access (similar to what owners can do but with some rights limited).  

As I understand you suggest that a team is automatically added because the repo creator belongs to it. Imho it adds a lot of complexity. If the creator belongs to two teams should both be added, or should it be possible to select?

Reply:
> 
> 
> shouldn't we also check if user is part of any team with admin rights?

@lafriks I think it's the point of this PR *not* to need that.